main.cpp

```diff
@@ -73,29 +78,37 @@ struct <#structName1#>                                //4
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float uVal1 { 0 }, uVal2 { 0 };
+    float membFunc(float* updVal)      //12
     {
-        
+        std::cout << "U's uVal1 value: " << this->uVal1 << std::endl;
+        this->uVal1 = *updVal;
+        std::cout << "U's uVal1 updated value: " << this->uVal1 << std::endl;
+        while( std::abs(this->uVal2 - this->uVal1) > 0.001f )
+        {
+            this->uVal2 += 1;        
+        }
+        std::cout << "U's uVal2 updated value: " << this->uVal2 << std::endl;
+        return this->uVal2 * this->uVal1;
     }
 };
 
-struct <#structname2#>
+struct W
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float statFunc(U* that, float updtdVal )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        std::cout << "U's uVal1 value: " << that->uVal1 << std::endl;
+        that->uVal1 = updtdVal;
```

Comment:
What do we always check before using pointers?
The placeholder for the 2nd argument to this function was a pointer.  
Make the 2nd argument a pointer.
- [ ] revise

Reply:
woops, my mistake. thanks